salt/state.py

```diff
@@ -3284,12 +3311,11 @@ def _read_tgt(tgt):
                             if tgt not in top[saltenv]:
                                 top[saltenv][tgt] = ctop[saltenv][tgt]
                                 continue
-                            m_type1, m_states1 = _read_tgt(top[saltenv][tgt])
-                            m_type2, m_states2 = _read_tgt(ctop[saltenv][tgt])
+                            kargs1, m_states_1 = _read_tgt(top[saltenv][tgt])
+                            kargs2, m_states_2 = _read_tgt(ctop[saltenv][tgt])
                             merged = []
-                            match_type = m_type2 or m_type1
-                            if match_type is not None:
-                                merged.append(match_type)
+                            merged.extend(kargs1)
+                            merged.extend(kargs2)
```

Comment:
The whole point of the three lines you're replacing with the above two lines, is to only append the `match_type` dict if it is specified. Hence the if statement, which only appends the dict if a `match` keyword param was specified. Appending twice here doesn't seem right to me, can you explain your thought process here? It would seem to me that if you wanted to support multiple keyword parameters, you'd want to do something like this:

```python
kargs = {}
kargs.update(kargs1)
kargs.update(kargs2)
if kargs:
    merged.append(kargs)
```

This would only append if at least one keyword param was used.

Then again, I may be misunderstanding, so I'd like to know what your intent was here. Another thing that doesn't make sense to me is that extending a list with a dict only gives you the key names:

```python
>>> x = []
>>> y = {'foo': 'bar', 'abc': 'def'}
>>> x.extend(y)
>>> x
['foo', 'abc']
>>>
```
So it would seem like we lose the actual values here.